sources/lib/structures/ty.mli

```diff
@@ -76,15 +79,25 @@ and trecord = {
   mutable lbs :  (Hstring.t * t) list;
   (** List of fields of the record. Each field has a name,
       and an associated type. *)
+  constr : Hstring.t; (* for ADTs that become records. default is "{" *)
 }
 (** Record types. *)
 
+type adt_constr =
+  (*  { constr : Hstring.t ; destrs : (Hstring.t * t) list }*)
+  Hstring.t * (Hstring.t * t) list
```

Comment:
The record option would be better than a tuple I think.
Also, maybe add some doc to explain what each field of the tuple stores ?

Reply:
ok